File: Backendd/ANIMO_Backend/src/main/java/edu/isep/animo_backend/models/Farm.java
package edu.isep.animo_backend.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "tblFarms")
public class Farm {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private @Getter @Setter Long id;
    private @Getter @Setter String name;

    @OneToOne
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    private @Getter @Setter Address address;

}

File: Frontend/ANIMO/src/app/modules/shared/models/farm.ts
import { Address } from "./address";

export class Farm {
    id: number;
    name: string;
    address: Address;
  
    constructor(id: number, name: string, address: Address) {
      this.id = id;
      this.name = name;
      this.address = address;
    }
  }
  